server/app.js

```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
+        });
+        sock.on('data', function(data) {
+            //console.log(data);
+            //Packet Type
+            var packetType = data.slice(0,1).toString();
+            packetType = "0x" + ascii_to_hexa(packetType);
+            packetType = parseInt(packetType);
+            //console.log("Packet Type: ", packetType);
+            //Payload Length
+            var payloadLength = data.slice(1,2).toString();
+            payloadLength = "0x" + ascii_to_hexa(payloadLength);
+            payloadLength = parseInt(payloadLength);
+            //console.log("Payload Len: ", payloadLength);
+            //Payload
+            var payload = data.slice(2,payload).toString();
+            //console.log(payload);
+            var payload = payload.split(":");
+
+            //console.log("Payload: ", payload[0]);
+            //console.log("Payload: ", payload[1]);
+            //console.log("Payload: ", payload[2]);
+            console.log(packetType , payloadLength, payload);            
+            
+            if(packetType == 0 )
+            {
+                socket.broadcast.emit('infoDiag', payload);
+            }
+            if(packetType == 1)
+            {
+                socket.broadcast.emit('errorDiag', payload);
+            }
+            if(packetType == 2)
+            {
+                socket.broadcast.emit('statusDiag', payload);
+            }
+            if(packetType == 3)
+            {
+                socket.broadcast.emit('log', payload);
+            }
+            //var payload = data.slice(,1).toString(); 
+            //var result = message.split(":");
+            //console.log(result[0]);
+            //console.log(result[1]);
+            //console.log(result[2]);
+
+
+            //byte = "0x" + ascii_to_hexa(byte);
+            //byte = parseInt(byte);
+            //socket.end(eval(data));
+        });
+    });
+
+    server.listen(5002, '192.168.43.12'); 
+    ///////////////////////////////////////////////////////////////// 
+
+
     var connected = false;
-	socket.on('stream',function(image){
-		socket.broadcast.emit('stream', image);
-	});
-	socket.on('event',function(data){
-		console.log('A client sent us this SMART message:', data.message);
-	});
-    socket.on('exit', function(ignore) {
+    socket.on('stream',function(image){
+        socket.broadcast.emit('stream', image);
+    });
+    socket.on('err', function(err){
+            // handle the error safely
+            console.log(err)
+        });
+    socket.on('event',function(data){
+        
+        if(data.message == 20) {
+            clearInterval(repeat);
+            data.message = 14;
+        }
+
+        client.connect(5000, "192.168.43.250", function() { //192.168.43.250 port 5000
+            console.log('Connected to 192.168.43.250'); 
+            console.log('Local client sending:', data.message);
+            buf = new Buffer([data.message, 0xA, 0xFF]); // 0xA is temp value
```

Comment:
`0xA` and `0xFF` should be declared as constant values.
For example:
`const var temp = 0xA`
`const var dummy = 0xFF`

Reply:
0xA isn't the default for all the buffer values there , but 0xFF can be labeled as a const